media/js/firefox/family/animate-checkmark.es6.js

```diff
@@ -0,0 +1,42 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+let _observer;
+
+const isSupported = () => {
+    return (
+        'IntersectionObserver' in window &&
+        'IntersectionObserverEntry' in window &&
+        'intersectionRatio' in window.IntersectionObserverEntry.prototype &&
+        window.NodeList &&
+        NodeList.prototype.forEach
+    );
+};
+
+const createObserver = () => {
+    return new IntersectionObserver(
+        function (entries) {
+            entries.forEach(function (entry) {
+                // trigger animation by adding relevant class with animation styles
+                if (entry.isIntersecting) {
+                    entry.target.classList.add('animate-checkmark');
+                    // remove target observer after triggering animation
+                    _observer.unobserve(entry.target);
+                }
+            });
+        },
+        { threshold: '0.4' }
+    );
+};
+
+export const init = () => {
+    if (isSupported()) {
```

Comment:
on the fence about two things here:

1. in general, should we keep intersection observer isSupported function in a shared file? (will be used across pop-in, checkmark, and other animations)

2. for agreement checkbox drawing, should we avoid all animation if user prefers reduced motion? this one seemed OK to keep even when reduced motion preferred, feels like a fun, subtle touch

Reply:
As long as each JS script is in the same bundle, seems like an easy enough thing for each function to import?